mitodl/mit-open
# OCW course completeness score

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5258

### Description (What does it do?)
- Adds a `completeness` field to `LearningResource` with a default value of 1.0
- Calculates the value for OCW courses during ETL contentfile ingestion


### How can this be tested?
- Copy `OCW_` settings from RC to your backend env file and also set `SKIP_TIKA=True`
- Run `docker compose up`, a migration should complete
- Run the following mgmt commands (do NOT use the `--skip-contentfiles` option):
   ```
  ./manage.py backpopulate_ocw_data --course-name 6-622-power-electronics-spring-2023
  ./manage.py backpopulate_ocw_data --course-name 6-801-machine-vision-fall-2020
  ./manage.py backpopulate_ocw_data --course-name sp-248-neet-ways-of-thinking-fall-2023
   ```
- When done, the courses should have a `completeness` score of 1.0, 0.8, and 0.033333333
- Try the new management command with several different options, they should complete without errors.
  ```
  ./manage.py backpopulate_ocw_scores --course-name  6-801-machine-vision-fall-2020
  ./manage.py backpopulate_ocw_scores  --csv --skip_calc scores.csv
  ```
